truffle/README.md

```diff
@@ -0,0 +1,220 @@
+
+This tutorial aims to help you get started with developing decentralized applications on [Optimism](https://optimism.io/) using HardHat. Applications 
+running on top of Optimism are about as secure as those running on the underlying Ethereum mainnet itself, but are
+[significantly cheaper](https://optimism.io/gas-comparison).
+
+
+## Build an Optimism Server
+
+To test and debug on Optimism you need to have a running Optimism server, so the first step is to build one. The directions in this section are 
+for an Ubuntu 20.04 VM running on GCP with a 20 GB disk (the default, 10 GB, is not enough), but they should be similar for other Linux 
+versions and other platforms
+
+### Install Prerequisite Software
+
+1. Install [Docker](https://www.docker.com/). If you prefer not to use the convenience script, [read the documentation
+   to learn other methods to install](https://docs.docker.com/engine/install/ubuntu).
+
+   ```sh
+   curl -fsSL https://get.docker.com -o get-docker.sh
+   sudo sh get-docker.sh
+   sudo apt install -y docker-compose
+   sudo usermod -a -G docker `whoami`
+   ```
+ 
+
+2. Install [Node.js](https://nodejs.org/en/) and a number of npm packages. The version in the OS repository is 
+  out of date, so we'll get the package from a different source.
+
+   ```sh
+   curl -sL https://deb.nodesource.com/setup_12.x -o nodesource_setup.sh
+   sudo bash nodesource_setup.sh
+   sudo apt install -y nodejs
+   sudo npm install -g yarn truffle ganache-cli
+   ```
+   
+3. Start a new terminal window to ensure the OS knows you are now a member of the `docker` 
+   group and authorized to create and manage Docker images and containers.
+
+
+### Start an Optimism Server
+
+This process downloads, compiles, and builds an Optimism network. Note that it takes a long time.
+
+```sh
+git clone https://github.com/ethereum-optimism/optimism.git
+cd optimism
+yarn install
+yarn build
+cd ops
+docker-compose build
+```
+
+Note that you will see a **Done** message at some point during the build process. Ignore it,
+it means that a specific section is done, and even though you do not see progress at that 
+moment the build is continuing.
+
+```sh
+docker-compose up
+```
+
+When start seeing log entries scrolling on the console it means the system is now running. 
+
+
+## Migrate a Dapp to Optimism
+
+Now that we have Optimism running, it is time to run a distributed application (dapp) on it.
+
+**Note:** If you don't need the explanations and just want to see running code, 
+[click here](https://github.com/ethereum-optimism/optimism-tutorial/). The 
+`truffle/dapp` directory
+is just an `npm install` away from being a working example.
+
+### Get a Sample Application
+
+The easiest way is to start with a sample application. 
+
+1. Open a second command line terminal
+2. Run `hardhat`, the development environment we use in this tutorial
```

Comment:
Seems like this is still using the content for `hardhat` instead of `truffle`

Reply:
Yes, I forgot I already submitted the PR. I'll move it to a different branch.